src/ray/common/id.cc

```diff
@@ -322,6 +315,33 @@ JobID PlacementGroupID::JobId() const {
       reinterpret_cast<const char *>(this->Data() + kUniqueBytesLength), JobID::kLength));
 }
 
+LeaseID LeaseID::FromRandom() {
+  std::string data(kUniqueBytesLength, 0);
+  uint32_t current_counter = counter_.fetch_add(1);
+  std::memcpy(data.data(), &current_counter, sizeof(current_counter));
+  std::copy_n(WorkerID::FromRandom().Data(), kUniqueIDSize, std::back_inserter(data));
+  return LeaseID::FromBinary(data);
+}
+
+LeaseID LeaseID::FromWorkerId(const WorkerID &worker_id) {
```

Comment:
I think it's better to pass in the counter `LeaseID::FromWorker(const WorkerID &worker_id, size_t counter)` similar to

```
TaskID TaskID::ForNormalTask(const JobID &job_id,
                             const TaskID &parent_task_id,
                             size_t parent_task_counter) {
```

and let the caller manage the counter. This way this function becomes a pure function without side-effect.